modules/rostests/apitests/user32/SystemParametersInfo.c

```diff
@@ -106,11 +106,130 @@ MSG_ENTRY CaptionHeight_chain[]={
 static void Test_NonClientMetrics()
 {
     NONCLIENTMETRICS NonClientMetrics;
+    NONCLIENTMETRICSA ncmA;
+    NONCLIENTMETRICSW ncmW;
+    BOOL ret;
 
     /* WARNING: this test requires themes and dwm to be disabled */
 
     SetCursorPos(0,0);
 
+    ncmA.cbSize = 0;
+    ret = SystemParametersInfo(SPI_GETNONCLIENTMETRICS, 0, &ncmA, 0);
+    ok_int(ret, 0);
+    FlushMessages();
+    COMPARE_CACHE(empty_chain);
+
+    ncmA.cbSize = sizeof(ncmA);
```

Comment:
I would suggest storing the test data into an array, and then looping on the array, calling `SystemParametersInfoA` or SystemParametersInfoW` depending on whether an ANSI or unicode systemparametersinfo is tested.